### ISSUE TITLE
Petinventory clear after switching Server

### ISSUE THREAD

**Issue by Serial1990** [MyPetORG/MyPet#1692]
# Bug report

* [x] I have checked other issues to make sure this is not a duplicate.

## Describe the bug

_A clear and concise description of what the bug is._

## To Reproduce

_Steps to reproduce the behavior, please provide code snippets or a repository:_

1. idk... Many renamed items in Shulkers.
...

## Expected behavior

Petinventory clear after Server Switch or Relog.

## Actual behaviour
Inventory bevor switching Server:
![image](https://github.com/user-attachments/assets/162124dd-c241-4a10-a96e-d28b03397fde)
Inventory after switching Server:
![image](https://github.com/user-attachments/assets/fe496a68-0057-484e-b711-28ccb1d540e6)

## Additional context
[ticket.zip](https://github.com/user-attachments/files/17636837/ticket.zip)

**Bukkit/Spigot/Bungee Build/Version: 1.21.1-123**

**MyPet Build/Version: 3.12-SNAPSHOT-B1768**

**Error:**
```
[19:11:13 WARN]: [de.Keyle.MyPet.MyPetPlugin] [MyPet] Plugin MyPet v3.12 generated an exception while executing task 2344314
java.lang.IllegalStateException: Cannot encode empty ItemStack
        at net.minecraft.world.item.ItemStack.save(ItemStack.java:620) ~[paper-1.21.1.jar:1.21.1-123-aebbdc9]
        at MyPet-3.12-SNAPSHOT-B1768.jar/de.Keyle.MyPet.compat.v1_21_R1.util.NBTHelper.save(NBTHelper.java:35) ~[MyPet-3.12-SNAPSHOT-B1768.jar:?]
        at MyPet-3.12-SNAPSHOT-B1768.jar/de.Keyle.MyPet.compat.v1_21_R1.util.inventory.ItemStackNBTConverter.itemStackToCompound(ItemStackNBTConverter.java:48) ~[MyPet-3.12-SNAPSHOT-B1768.jar:?]
        at MyPet-3.12-SNAPSHOT-B1768.jar/de.Keyle.MyPet.compat.v1_21_R1.util.inventory.CustomInventory.save(CustomInventory.java:237) ~[MyPet-3.12-SNAPSHOT-B1768.jar:?]
        at MyPet-3.12-SNAPSHOT-B1768.jar/de.Keyle.MyPet.skill.skills.BackpackImpl.save(BackpackImpl.java:115) ~[MyPet-3.12-SNAPSHOT-B1768.jar:?]
        at MyPet-3.12-SNAPSHOT-B1768.jar/de.Keyle.MyPet.entity.MyPet.getSkillInfo(MyPet.java:336) ~[MyPet-3.12-SNAPSHOT-B1768.jar:?]
        at MyPet-3.12-SNAPSHOT-B1768.jar/de.Keyle.MyPet.repository.types.MySqlRepository$9.run(MySqlRepository.java:803) ~[MyPet-3.12-SNAPSHOT-B1768.jar:?]
        at org.bukkit.craftbukkit.scheduler.CraftTask.run(CraftTask.java:86) ~[paper-1.21.1.jar:1.21.1-123-aebbdc9]
        at org.bukkit.craftbukkit.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:57) ~[paper-1.21.1.jar:1.21.1-123-aebbdc9]
        at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[paper-1.21.1.jar:?]
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
```

**Comment by Jakllp** [comment]
Welcome to Mojang changing pretty much everything about items.  
Thanks for the report

**Comment by Serial1990** [comment]
> Welcome to Mojang changing pretty much everything about items. Thanks for the report

Yes, I know... I feel really sorry for you too...

What I noticed is that it's not a specific item that causes the problem, but the amount of items.

**Comment by Serial1990** [comment]
We are currently having huge problems with the sync of inventories/effects and pet experiences.

We are constantly getting messages that the effects have been reset when changing servers or that items in the inventories have disappeared.

I also suspect that it is some kind of sync problem.

**Comment by Jakllp** [comment]
To give an update on this:  
I believe I have identified the problem and mitigated it.  
@MaksyKun has a test-jar and will be testing if they actually work.

**Comment by Serial1990** [comment]
> To give an update on this: I believe I have identified the problem and mitigated it. @MaksyKun has a test-jar and will be testing if they actually work.

If you like, I can test it too. :)

We always have around 50 players online, so it's a good place to test it.

**Comment by Jakllp** [comment]
Fixed in [3.12-SNAPSHOT-B1771](https://github.com/MyPetORG/MyPet/releases/tag/3.12-SNAPSHOT-B1771)
